persistence-api/src/main/java/io/stargate/db/ClientInfo.java

```diff
@@ -41,12 +45,34 @@ public void registerDriverInfo(DriverInfo info) {
     return Optional.ofNullable(driverInfo);
   }
 
-  public ByteBuffer getAuthenticatedUser() {
+  public AuthenticatedUser getAuthenticatedUser() {
     return authenticatedUser;
   }
 
-  public void setAuthenticatedUser(ByteBuffer authenticatedUser) {
+  public void setAuthenticatedUser(AuthenticatedUser authenticatedUser) {
     this.authenticatedUser = authenticatedUser;
+    this.token =
+        authenticatedUser.token() != null
+            ? ByteBuffer.wrap(authenticatedUser.token().getBytes(StandardCharsets.UTF_8))
+            : null;
+    this.roleName = ByteBuffer.wrap(authenticatedUser.name().getBytes(StandardCharsets.UTF_8));
+    this.isFromExternalAuth =
+        ByteBuffer.wrap(
+            (authenticatedUser.isFromExternalAuth()
+                ? new byte[] {(byte) 1}
+                : new byte[] {(byte) 0}));
```

Comment:
nit: Since we're are the only consumer of these buffers, we could probably use `null` to indicate `false` and a static constant buffer to indicate `true`.

Reply:
True, that cleans it up a little